telegram/request.py

```diff
@@ -382,17 +383,18 @@ def retrieve(self, url: str, timeout: float = None) -> bytes:
 
         return self._request_wrapper('GET', url, **urlopen_kwargs)
 
-    def download(self, url: str, filename: str, timeout: float = None) -> None:
+    def download(self, url: str, filepath: Union[Path, str], timeout: float = None) -> None:
         """Download a file by its URL.
 
         Args:
             url (:obj:`str`): The web location we want to retrieve.
+            filepath (:obj:`Path` | :obj: `str`): The filepath to download the file to.
             timeout (:obj:`int` | :obj:`float`, optional): If this value is specified, use it as
                 the read timeout from the server (instead of the one specified during creation of
                 the connection pool).
-            filename (:obj:`str`): The filename within the path to download the file.
+
+        .. versionchanged:: 14.0
+            The
```

Comment:
?

Reply:
fixed the doc.

enabled sending ```pathlib.Path``` objects here as ```filepath``` argument.
Do i need to add tests? if so, where?

Reply:
That would indeed be nice - we don't have a test for it so far at all IISC. Please add it to `test_request.py` and make sure to test both pathing a string and a `Path` :)

Reply:
added a parametrized ```test_download```